File: MiniProyectoPooCodigoPython/ec.edu.utpl.poo.semana8.proyecto/Motorcycle.py
# importamos date de datetime para obtener el año en que estamos para que el año de la moto no sea mayor al actual
from datetime import date

class Motorcycle:

    def __init__(self, brand, color, year):
        self.__brand = brand
        self.__color = color
        if (year > 1885 and year < date.today().year):
            self.__year = year
        else :
            raise Exception("El año de la motocycleta no puede ser antes de 1885 o mayor a este año")

    @property
    def brand (self):
        return self.__brand

    @brand.setter
    def brand(self, b):
        self.__brand = b

    @property
    def color(self):
        return self.__color

    @color.setter
    def color(self, c):
        self.__color = c

    @property
    def year(self):
        return self.__year

    @year.setter
    def year(self, y):
        if (y > 1885 and y < date.today().year):
            self.__year = y
        else :
            raise Exception("El año de la motocycleta no puede ser antes de 1885 o mayor a este año")

File: MiniProyectoPoo/src/ec/edu/utpl/poo/semana8/proyecto/Motorcycle.java
package ec.edu.utpl.poo.semana8.proyecto;
import java.time.LocalDate;
public class Motorcycle {

    // atributo de clase
    static int llantas = 2;

    // atributos de instancia
    private String brand;
    private String color;
    private int year;

    // constructor

    public Motorcycle(String brand, String color, int year) {
        this.brand = brand;
        this.color = color;
        if (year > 1885 && year < LocalDate.now().getYear()) {
            this.year = year;
        } else {
            throw new IllegalArgumentException("El año de la motocycleta no puede ser antes de 1885 o mayor a este año");
        }
    }

    // sobrecarga de constructores

    public Motorcycle(String brand) {
        this.brand = brand;
    }

    public Motorcycle(String brand, String color) {
        this.brand = brand;
        this.color = color;
    }

    public Motorcycle(String color, int year) {
        this.color = color;
        this.year = year;
    }

    public Motorcycle(int year) {
        this.year = year;
    }

    // metodos

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        if (year > 1885 && year < LocalDate.now().getYear()) {
            this.year = year;
        } else {
            throw new IllegalArgumentException("El año de la motocycleta no puede ser antes de 1885 o mayor a este año");
        }
    }
}